crypto/err/err_save.c

```diff
@@ -43,6 +43,71 @@ void OSSL_ERR_STATE_save(ERR_STATE *es)
     memset(thread_es, 0, sizeof(*thread_es));
 }
 
+void OSSL_ERR_STATE_save_to_mark(ERR_STATE *es)
+{
+    size_t i, j, count;
+    int top;
+    ERR_STATE *thread_es;
+
+    if (es == NULL)
+        return;
+
+    thread_es = ossl_err_get_state_int();
+    if (thread_es == NULL) {
+        for (i = 0; i < ERR_NUM_ERRORS; ++i)
+            err_clear(es, i, 1);
+
+        es->top = es->bottom = 0;
+        return;
+    }
+
+    /* Determine number of errors we are going to move. */
+    for (count = 0, top = thread_es->top;
+         thread_es->bottom != top
+         && thread_es->err_marks[top] == 0;
+         ++count)
+        top = top > 0 ? top - 1 : ERR_NUM_ERRORS - 1;
```

Comment:
Seems virtually identical to ERR_count_to_mark(). Is there no opportunity for code reuse?

Reply:
I might factor this in a subsequent PR. Not doing this now so we can get this in.